generation/maisi/scripts/compute_fid_2-5d_ct.py

```diff
@@ -0,0 +1,655 @@
+
+        images_2d = torch.cat(slices, dim=0)
+        images_2d = radimagenet_intensity_normalisation(images_2d)
+        images_2d = images_2d[mapping_index]
+
+        feature_image_zx = feature_network.forward(images_2d)
+        feature_image_zx = spatial_average(feature_image_zx, keepdim=False)
+
+    return feature_image_xy, feature_image_yz, feature_image_zx
+
+
+def pad_to_max_size(tensor: torch.Tensor, max_size: int, padding_value: float = 0.0) -> torch.Tensor:
+    """
+    Zero-pad a 2D feature map or other tensor along the first dimension to match a specified size.
+
+    Args:
+        tensor (torch.Tensor): The feature tensor to pad.
+        max_size (int): Desired size along the first dimension.
+        padding_value (float): Value to fill during padding.
+
+    Returns:
+        torch.Tensor: Padded tensor matching `max_size` along dim=0.
+    """
+    pad_size = [0, 0] * (len(tensor.shape) - 1) + [0, max_size - tensor.shape[0]]
+    return F.pad(tensor, pad_size, "constant", padding_value)
+
+
+def main(
+    data0_dataroot: str = "path/to/datasetA",
+    data0_filelist: str = "path/to/filelistA.txt",
+    data0_folder: str = "datasetA",
+    data1_dataroot: str = "path/to/datasetB",
+    data1_filelist: str = "path/to/filelistB.txt",
+    data1_folder: str = "datasetB",
+    enable_center_slices: bool = False,
+    enable_center_slices_ratio: float = 0.4,
+    enable_padding: bool = True,
+    enable_center_cropping: bool = True,
+    enable_resampling: bool = False,
+    enable_resampling_spacing: str = "1.0x1.0x1.0",
+    ignore_existing: bool = False,
+    model_name: str = "radimagenet_resnet50",
+    num_images: int = 100,
+    output_root: str = "./features/features-512x512x512",
+    target_shape: str = "512x512x512",
+):
+    """
+    Main function to compute 2.5D features for two datasets (e.g. real vs. synthetic)
+    and calculate FID across XY, YZ, ZX planes.
+
+    Args:
+        data0_dataroot (str): Root path of dataset 0 (real images).
+        data0_filelist (str): Text file listing the 3D images in dataset 0.
+        data0_folder (str): Name (subdir) for dataset 0 outputs under `output_root`.
+        data1_dataroot (str): Root path of dataset 1 (synthetic images).
+        data1_filelist (str): Text file listing the 3D images in dataset 1.
+        data1_folder (str): Name (subdir) for dataset 1 outputs under `output_root`.
+        enable_center_slices (bool): If True, only slices around the center ratio are used.
+        enable_center_slices_ratio (float): Ratio of center slices to keep if `enable_center_slices` is True.
+        enable_padding (bool): Whether to pad images to `target_shape`.
+        enable_center_cropping (bool): Whether to center-crop images to `target_shape`.
+        enable_resampling (bool): Whether to resample images to `enable_resampling_spacing`.
+        enable_resampling_spacing (str): Target voxel spacing as "XxYxZ", e.g. "1.0x1.0x1.0".
```

Comment:
I would suggestion we combine the bool arg within the `enable_resampling_spacing` and `enable_center_slices_ratio`. Such as if enable_center_slices_ratio is not None, then it means enable center slices.
This may can make less arguments, what do you think?

Reply:
Good point. I made changes to reduce the redundancy.